### PR TITLE
#4453 Update .pre-commit-config.yaml and removed unused imports

### PR DESCRIPTION
**Pull request by asinghgaba**
**Thank your for opening a PR!**

Depending on what your PR does, here are a few things you might want to address in the description:
+ [x] what are the (breaking) changes that this PR makes?
+ [x] right before it's ready to merge, mention the PR in the RELEASE-NOTES.md /not required

Completed all the tasks raised in issue #4453

### CODE REVIEW

**Review by MarcoGorelli — APPROVED** [review]
Thanks @asinghgaba , looks good to me!

**Comment by asinghgaba** [comment]
Great @MarcoGorelli this was my first ever official PR! Looking forward to contributing more and learning new things.

File: pymc3/distributions/posterior_predictive.py [hunk 1]
```diff
@@ -69,15 +57,15 @@ class _TraceDict(UserDict):
...
     data: Point
 
     def __init__(
         self,
-        point_list: Optional[List[Point]] = None,
-        multi_trace: Optional[MultiTrace] = None,
-        dict_: Optional[Point] = None,
+        point_list: list[Point] | None = None,
+        multi_trace: MultiTrace | None = None,
+        dict_: Point | None = None,
```
Comment (michaelosthege) [inline pymc3/distributions/posterior_predictive.py:68]: This syntactic sugar is new to me - since which Python version is this supported?

**Review by michaelosthege — APPROVED** [review]

File: pymc3/distributions/posterior_predictive.py (hunk 1 above)
Comment (asinghgaba) [reply pymc3/distributions/posterior_predictive.py:68]: Yeah, the code looks much cleaner now!
I'm not sure but I'm guessing maybe 3.9.

Comment (michaelosthege) [reply pymc3/distributions/posterior_predictive.py:68]: PyMC3 supports Python >=3.7... The CI was fine, but please find out exactly then - we don't want to break something by accident.

Comment (asinghgaba) [reply pymc3/distributions/posterior_predictive.py:68]: Okay, I'll check this now and let you know.

Comment (asinghgaba) [reply pymc3/distributions/posterior_predictive.py:68]: Hey @michaelosthege I just ran the test for posterior_predicitive  on my version of pymc3, using python 3.7, and everything works fine.

Comment (MarcoGorelli) [reply pymc3/distributions/posterior_predictive.py:68]: It works fine in Python3+ as long as there's `from __future__ import annotations` at the top (and `pyupgrade` will only use this syntax in files where `from __future__ import annotations` is present)

Comment (MarcoGorelli) [reply pymc3/distributions/posterior_predictive.py:68]: In 3.10 it'll become the default and'll work without `from __future__ import annotations`

Comment (michaelosthege) [reply pymc3/distributions/posterior_predictive.py:68]: We're good to go then. Thanks!

**Pull request merged**